homeassistant/components/neato/config_flow.py

```diff
@@ -0,0 +1,119 @@
+"""Config flow to configure Neato integration."""
+
+import logging
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import callback
+
+from .const import CONF_VENDOR, NEATO_DOMAIN, VALID_VENDORS
+
+
+DOCS_URL = "https://www.home-assistant.io/components/neato"
+DEFAULT_VENDOR = "neato"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def configured_neato(hass):
+    """Return the configured Neato Account."""
+    entries = hass.config_entries.async_entries(NEATO_DOMAIN)
+    if entries:
+        return entries[0]
+    return None
+
+
+class NeatoConfigFlow(config_entries.ConfigFlow, domain=NEATO_DOMAIN):
+    """Neato integration config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize flow."""
+        self._username = vol.UNDEFINED
+        self._password = vol.UNDEFINED
+        self._vendor = vol.UNDEFINED
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if configured_neato(self.hass) is not None:
```

Comment:
ConfigFlow has a helper method, you can do:

```suggestion
        if self._async_current_entries():
```

Reply:
done, that's a nice solution